### PR TITLE
Update Other Metagames Commands

### PR DESCRIPTION
**Pull request by Spandamn**

### CODE REVIEW

**Comment by xfix** [comment]
I'm not entirely sure why redirect to `/data`, but only sometimes? What was wrong with old code, this just seems more complicated for no real reason.

**Comment by Zarel** [comment]
`/tiershift` does need to be updated, but I don't think delegating to `/data` is the right choice.

File: chat-plugins/othermetas.js [hunk 1]
```diff
@@ -105,9 +105,10 @@ exports.commands = {
 		for (let i in pokeobj.baseStats) {
 			bst += pokeobj.baseStats[i];
 		}
+		if (bst > 350) return this.sendReply(`|html|${Chat.getDataPokemonHTML(pokeobj)}`);
 		let newStats = {};
 		for (let i in pokeobj.baseStats) {
-			newStats[i] = pokeobj.baseStats[i] * (bst <= 350 ? 2 : 1);
+			newStats[i] = pokeobj.baseStats[i] * 2;
```
Comment (Zarel) [inline chat-plugins/othermetas.js]: I still don't know why you think this is an improvement over the old code.

Comment (Spandamn) [inline chat-plugins/othermetas.js]: In the old one, the BST is being check multiple times since it's inside a loop

Comment (xfix) [inline chat-plugins/othermetas.js]: First of all [loop-invariant code motion optimization](https://en.wikipedia.org/wiki/Loop-invariant_code_motion) is a thing. Second, even if that optimization wasn't a thing, comparing to the cost of loop over a map (which happens just above), cost of comparing an integer (pretty much one CPU instruction, three considering conditional moves here) is irrelevant.

File: chat-plugins/othermetas.js [hunk 2]
```diff
@@ -133,7 +134,7 @@ exports.commands = {
 			'LC': 40,
 		};
 		let template = Object.assign({}, Dex.getTemplate(target));
-		if (!(template.tier in boosts)) return this.sendReplyBox(`${template.species} in Tier Shift: <br /> ${Object.values(template.baseStats).join('/')}`);
+		if (!(template.tier in boosts)) return this.sendReply(`|html|${Chat.getDataPokemonHTML(template)}`);
```
Comment (xfix) [inline chat-plugins/othermetas.js]: I think just having that single line change would be fine and accepted however.

File: chat-plugins/othermetas.js (hunk 1 above)
Comment (Zarel) [inline chat-plugins/othermetas.js]: Yes, DRYer code is better than this sort of optimization.

Comment (Spandamn) [inline chat-plugins/othermetas.js]: okay, will revert back

**Comment by Spandamn** [comment]
@Zarel @TheImmortal

File: chat-plugins/othermetas.js [hunk 3]
```diff
@@ -7,16 +7,10 @@ exports.commands = {
...
-		let template = toId(sep[0]);
-		if (!Dex.data.Items[stone] || (Dex.data.Items[stone] && !Dex.data.Items[stone].megaEvolves && !Dex.data.Items[stone].onPrimal)) {
-			return this.errorReply(`Error: Mega Stone not found`);
-		}
-		if (!Dex.data.Pokedex[toId(template)]) {
-			return this.errorReply(`Error: Pokemon not found`);
-		}
-		template = Object.assign({}, Dex.getTemplate(template));
-		stone = Object.assign({}, Dex.getItem(stone));
+		let stone = Object.assign({}, Dex.getItem(sep[1]));
```
Comment (Zarel) [inline chat-plugins/othermetas.js]: There's no need to shallow clone `stone`; it's not modified at all.

**Comment by Spandamn** [comment]
Bumping again @Zarel @TheImmortal

**Pull request merged**